File: 2022-2023/sourcecode/lab/lab9/ROOMbasicsJava/app/src/main/java/pl/udu/uwr/pum/roombasicsjava/view/WordsComparator.java
package pl.udu.uwr.pum.roombasicsjava.view;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import pl.udu.uwr.pum.roombasicsjava.model.Word;

class WordsComparator extends DiffUtil.ItemCallback<Word> {

    @Override
    public boolean areItemsTheSame(@NonNull Word oldItem, @NonNull Word newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(@NonNull Word oldItem, @NonNull Word newItem) {
        return oldItem.getWord().equals(newItem.getWord());
    }
}

File: 2022-2023/sourcecode/lab/lab9/ROOMbasicsKotlin/app/src/main/java/pl/udu/uwr/pum/roombasicskotlin/view/WordsComparator.kt
package pl.udu.uwr.pum.roombasicskotlin.view

import androidx.recyclerview.widget.DiffUtil
import pl.udu.uwr.pum.roombasicskotlin.model.Word

class WordsComparator : DiffUtil.ItemCallback<Word>() {
    override fun areItemsTheSame(oldItem: Word, newItem: Word): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: Word, newItem: Word): Boolean {
        return oldItem.word == newItem.word
    }
}